src/main/java/org/hyperledger/besu/riscv/poc/evm/BlockRunner.java

```diff
@@ -0,0 +1,431 @@
+            // no nop
+          }
+
+          @Override
+          public <T extends BesuService> Optional<T> getService(Class<T> serviceType) {
+            return Optional.empty();
+          }
+        };
+
+    final BonsaiWorldStateKeyValueStorage.Updater stateUpdater = bonsaiStorage.updater();
+    trienodes.forEach(
+        (hash, value) -> {
+          stateUpdater
+              .getWorldStateTransaction()
+              .put(KeyValueSegmentIdentifier.TRIE_BRANCH_STORAGE, hash.toArray(), value.toArray());
+        });
+    codes.forEach(
+        (hash, value) -> {
+          stateUpdater
+              .getWorldStateTransaction()
+              .put(
+                  KeyValueSegmentIdentifier.CODE_STORAGE,
+                  hash.toArrayUnsafe(),
+                  value.toArrayUnsafe());
+        });
+    final BlockHeader head = prevHeaders.get(prevHeaders.size() - 1);
+    stateUpdater
+        .getWorldStateTransaction()
+        .put(
+            KeyValueSegmentIdentifier.TRIE_BRANCH_STORAGE,
+            WORLD_ROOT_HASH_KEY,
+            head.getStateRoot().toArray());
+    stateUpdater
+        .getWorldStateTransaction()
+        .put(
+            KeyValueSegmentIdentifier.TRIE_BRANCH_STORAGE,
+            WORLD_BLOCK_HASH_KEY,
+            head.getBlockHash().toArray());
+    stateUpdater.commit();
+
+    var worldStateArchive =
+        new BonsaiWorldStateProvider(
+            bonsaiStorage,
+            blockchain,
+            Optional.empty(), // maxLayersToLoad
+            new NoopBonsaiCachedMerkleTrieLoader(),
+            serviceManager,
+            evmConfiguration,
+            () -> (__, ___) -> {},
+            new CodeCache());
+
+    /*
+    curl --location 'http://127.0.0.1:8545' \
+    --data '{
+        "jsonrpc": "2.0",
+        "method": "debug_traceTransaction",
+        "params": [
+            "0xca36c1e529714639c25fc2ac9a3b13c4d0a3e81a74f451313ec2926c54cc48f3",
+            {
+                "disableStorage": true
+            }
+        ],
+        "id": 1
+    }'
+     */
+    var postMergeContext = new PostMergeContext();
+
+    var protocolContext =
+        new ProtocolContext.Builder()
+            .withBlockchain(blockchain)
+            .withWorldStateArchive(worldStateArchive)
+            .withConsensusContext(postMergeContext)
+            .withBadBlockManager(new BadBlockManager())
+            .withServiceManager(serviceManager)
+            .build();
+
+    return new BlockRunner(protocolSchedule, protocolContext, blockchain, worldStateArchive);
+  }
+
+  private BlockRunner(
+      ProtocolSchedule protocolSchedule,
+      ProtocolContext protocolContext,
+      MutableBlockchain blockchain,
+      BonsaiWorldStateProvider worldStateArchive) {
+    this.protocolSchedule = protocolSchedule;
+    this.protocolContext = protocolContext;
+    this.blockchain = blockchain;
+    this.worldStateArchive = worldStateArchive;
+  }
+
+  public void processBlock(final Block block) {
```

Comment:
You are using a wrong version of the BlockRunner. I modified some stuff on the class. Could you use the last version I merged and apply your modification ?